packages/content/lib/content-sources/sanity-source.js

```diff
@@ -0,0 +1,273 @@
+/**
+ * @module sanity-source
+ */
+
+import chalk from 'chalk';
+import jsonpath from 'jsonpath';
+
+import ContentSource, { SourceOptions } from './content-source.js';
+import ContentResult from './content-result.js';
+import Credentials from '../credentials.js';
+import { Logger } from '@bluecadet/launchpad-utils';
+
+import sanityClient from '@sanity/client';
+import { toHTML } from '@portabletext/to-html';
+import toMarkdown from '@sanity/block-content-to-markdown';
+
+/**
+ * Options for SanitySource
+ */
+export class SanityOptions extends SourceOptions {
+	constructor({
+		apiVersion = "v2021-10-21",
+		projectid = undefined,
+		dataset = "production",
+		apiToken = undefined,
+		useCdn = false,
+    baseUrl = undefined,
+		queries = [],
+		textConverters = [],
+		limit = 100,
+		maxNumPages = -1,
```

Comment:
Have you tested various pagination scenarios? Speaking from my own experience that's often an area that breaks much later down the road once you reach a significant amount of content 😅